src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,832 @@
+									}
+								}
+								else if (childNode.IsExpanded)
+								{
+									allOpenedDescendantsCount += CountDescendants(childNode);
+								}
+							}
+						}
+
+						break;
+					}
+
+				// Removes a node from the ViewModel when a TreeNode
+				// removes a child.
+				case (CollectionChange.ItemRemoved):
+					{
+						var removingNodeParent = sender;
+						if (removingNodeParent.IsExpanded)
+						{
+							var removedNode = GetRemovedChildTreeViewNodeByIndex(removingNodeParent, index);
+							RemoveNodeAndDescendantsFromView(removedNode);
+							if (IsContentMode)
+							{
+								m_itemToNodeMap.Remove(removedNode.Content);
+							}
+						}
+
+						break;
+					}
+
+				// Triggered by a replace such as SetAt.
+				// Updates the TreeNode that changed in the ViewModel.
+				case (CollectionChange.ItemChanged):
+					{
+						var targetNode = sender.Children[index];
+						var changingNodeParent = sender;
+						if (changingNodeParent.IsExpanded)
+						{
+							var removedNode = GetRemovedChildTreeViewNodeByIndex(changingNodeParent, index);
+							if (!IndexOfNode(removedNode, out var removedNodeIndex))
+							{
+								throw new InvalidOperationException("Node does not exist");
+							}
+
+							RemoveNodeAndDescendantsFromView(removedNode);
+							Insert(removedNodeIndex, targetNode);
+
+							if (IsContentMode)
+							{
+								m_itemToNodeMap.Remove(removedNode.Content);
+								m_itemToNodeMap.Add(targetNode.Content, targetNode);
+							}
+						}
+
+						break;
+					}
+			}
+		}
+
+		private void SelectedNodeChildrenChanged(TreeViewNode sender, object args)
+		{
+			var vectorChangedArgs = (IVectorChangedEventArgs)args;
+			CollectionChange collectionChange = vectorChangedArgs.CollectionChange;
+			var index = (int)vectorChangedArgs.Index; // Cast to int to be able to index
+			var changingChildrenNode = (TreeViewNode)sender;
+
+			switch (collectionChange)
+			{
+				case (CollectionChange.ItemInserted):
+					{
+						var newNode = changingChildrenNode.Children[index];
+						UpdateNodeSelection(newNode, NodeSelectionState(changingChildrenNode));
+						break;
+					}
+
+				case (CollectionChange.ItemChanged):
+					{
+						var newNode = changingChildrenNode.Children[index];
+						UpdateNodeSelection(newNode, NodeSelectionState(changingChildrenNode));
+
+						var selectedNodes = m_selectedNodes;
+						for (var i = 0; i < selectedNodes.Count; i++)
+						{
+							var selectNode = selectedNodes[i];
+							var ancestorNode = selectNode.Parent;
+							while (ancestorNode != null && ancestorNode.Parent != null)
+							{
+								ancestorNode = ancestorNode.Parent;
+							}
+
+							if (ancestorNode != m_originNode)
+							{
+								selectedNodes.RemoveAtCore(i);
+								//TODO
+								selectNode.ChildrenChanged -= SelectedNodeChildrenChanged;
+								//m_selectedNodeChildrenChangedEventTokenVector.erase(m_selectedNodeChildrenChangedEventTokenVector.begin() + i);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)